File: common/src/main/java/com/leo/commonutil/app/SystemShareHelp.java
package com.leo.commonutil.app;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.support.annotation.NonNull;
import android.support.v4.content.FileProvider;
import android.text.TextUtils;

import java.io.File;
import java.util.ArrayList;

public class SystemShareHelp {
    /**
     * 分享文件
     */
    public static void shareFile(@NonNull String title, @NonNull String filePath) {
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_SEND);
        intent.addCategory("android.intent.category.DEFAULT");
        Uri pdfUri;
        pdfUri = Uri.fromFile(new File(filePath));
        intent.putExtra(Intent.EXTRA_STREAM, pdfUri);
        intent.setType("application/pdf");
        try {
            AppStackManager.getInstance().getCurrentActivity()
                    .startActivity(Intent.createChooser(intent, title));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    //分享文字
    public static void shareText(@NonNull String shareText) {
        Intent shareIntent = new Intent();
        shareIntent.setAction(Intent.ACTION_SEND);
        shareIntent.putExtra(Intent.EXTRA_TEXT, shareText);
        shareIntent.setType("text/plain");
        try {
            AppStackManager.getInstance().getCurrentActivity()
                    .startActivity(Intent.createChooser(shareIntent, "分享到"));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    /**
     * 分享单张图片
     *
     * @param imagePath
     */
    public static void shareImage(@NonNull String imagePath) {
        try {
            Activity activity = AppStackManager.getInstance().getCurrentActivity();
            //由文件得到uri
            Uri imageUri;
            File imgFile = new File(imagePath);
            //判断当前手机版本
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                imageUri = FileProvider.getUriForFile(activity, activity.getPackageName() + ".fileprovider", imgFile);
            } else {
                imageUri = Uri.fromFile(imgFile);
            }
            Intent shareIntent = new Intent();
            shareIntent.setAction(Intent.ACTION_SEND);
            shareIntent.putExtra(Intent.EXTRA_STREAM, imageUri);
            shareIntent.setType("image/*");
            activity
                    .startActivity(Intent.createChooser(shareIntent, "分享到"));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    /**
     * 分享多张图片
     *
     * @param imgPaths
     */
    public static void shareImages(@NonNull ArrayList<String> imgPaths) {
        try {
            Activity activity = AppStackManager.getInstance().getCurrentActivity();
            ArrayList<Uri> uriList = new ArrayList<>();
            for (String img : imgPaths) {
                if (!TextUtils.isEmpty(img)) {
                    Uri imageUri;
                    File imgFile = new File(img);
                    //判断当前手机版本
                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                        imageUri = FileProvider.getUriForFile(activity, activity.getPackageName() + ".fileprovider", imgFile);
                    } else {
                        imageUri = Uri.fromFile(imgFile);
                    }
                    uriList.add(imageUri);
                }
            }
            Intent shareIntent = new Intent();
            shareIntent.setAction(Intent.ACTION_SEND_MULTIPLE);
            shareIntent.putParcelableArrayListExtra(Intent.EXTRA_STREAM, uriList);
            shareIntent.setType("image/*");
            activity
                    .startActivity(Intent.createChooser(shareIntent, "分享到"));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: common/src/main/java/com/leo/commonutil/share/SystemShareHelp.kt
package com.leo.commonutil.share

import android.content.Intent
import android.net.Uri
import android.os.Build
import android.text.TextUtils
import androidx.core.content.FileProvider
import com.leo.commonutil.app.AppStackManager
import java.io.File
import java.util.*

object SystemShareHelp {
    /**
     * 分享文件
     */
    fun shareFile(title: String, filePath: String) {
        val intent = Intent()
        intent.action = Intent.ACTION_SEND
        intent.addCategory("android.intent.category.DEFAULT")
        val pdfUri = Uri.fromFile(File(filePath))
        intent.putExtra(Intent.EXTRA_STREAM, pdfUri)
        intent.type = "application/pdf"
        try {
            AppStackManager.instance.currentActivity!!
                    .startActivity(Intent.createChooser(intent, title))
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    /**
     * 分享文字
     * @param shareText
     */
    fun shareText(shareText: String) {
        val shareIntent = Intent()
        shareIntent.action = Intent.ACTION_SEND
        shareIntent.putExtra(Intent.EXTRA_TEXT, shareText)
        shareIntent.type = "text/plain"
        try {
            AppStackManager.instance.currentActivity!!
                    .startActivity(Intent.createChooser(shareIntent, "分享到"))
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    /**
     * 分享单张图片
     *
     * @param imagePath
     */
    fun shareImage(imagePath: String) {
        try {
            val activity = AppStackManager.instance.currentActivity
            val imgFile = File(imagePath)
            val imageUri = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                FileProvider.getUriForFile(activity!!, "${activity.packageName}.fileprovider", imgFile)
            } else {
                Uri.fromFile(imgFile)
            }
            val shareIntent = Intent()
            shareIntent.action = Intent.ACTION_SEND
            shareIntent.putExtra(Intent.EXTRA_STREAM, imageUri)
            shareIntent.type = "image/*"
            activity!!
                    .startActivity(Intent.createChooser(shareIntent, "分享到"))
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    /**
     * 分享多张图片
     *
     * @param imgPaths
     */
    fun shareImages(imgPaths: ArrayList<String>) {
        try {
            val activity = AppStackManager.instance.currentActivity
            val uriList = ArrayList<Uri>()
            for (img in imgPaths) {
                if (!TextUtils.isEmpty(img)) {
                    val imgFile = File(img)
                    val imageUri = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                        FileProvider.getUriForFile(activity!!, activity.packageName + ".fileprovider", imgFile)
                    } else {
                        Uri.fromFile(imgFile)
                    }
                    uriList.add(imageUri)
                }
            }
            val shareIntent = Intent()
            shareIntent.action = Intent.ACTION_SEND_MULTIPLE
            shareIntent.putParcelableArrayListExtra(Intent.EXTRA_STREAM, uriList)
            shareIntent.type = "image/*"
            activity!!.startActivity(Intent.createChooser(shareIntent, "分享到"))
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
